framework/src/modules/token/method.ts

```diff
@@ -706,8 +742,37 @@ export class TokenMethod extends BaseMethod {
 		this.events.get(TokenIDSupportedEvent).log(methodContext, tokenID);
 	}
 
-	public async removeSupportTokenID(methodContext: MethodContext, tokenID: Buffer): Promise<void> {
-		await this.stores.get(SupportedTokensStore).removeSupportForToken(methodContext, tokenID);
+	public async removeSupport(methodContext: MethodContext, tokenID: Buffer): Promise<void> {
+		const [chainID] = splitTokenID(tokenID);
+
+		const allTokensSupported = await this._isAllTokensSupported(methodContext);
+
+		if (allTokensSupported) {
+			throw new Error('All tokens are supported.');
+		}
+
+		if (tokenID.equals(this.getMainchainTokenID()) || chainID.equals(this._config.ownChainID)) {
+			throw new Error('Cannot remove support for the specified token.');
+		}
+
+		const supportedTokens = await this._getSupportedTokens(methodContext, chainID);
+
+		if (supportedTokens) {
```

Comment:
Given the comment above, this should be changed. This code would be run unless in line 758 there was not error due to not having a store entry. Again, I guess try-except is the way to go, but you know better than me.